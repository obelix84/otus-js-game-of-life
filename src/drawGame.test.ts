/**
 * @jest-environment jsdom
 * */

import { drawGameOnCanvas } from "./drawGame";

describe("Write cells on canvas", () => {
  let canvas;
  let ctx: CanvasRenderingContext2D;

  beforeAll(() => {
    canvas = document.createElement("canvas");
    canvas.id = "canvas";
    canvas.width = 500;
    canvas.height = 500;
    document.body.appendChild(canvas);
    ctx = <CanvasRenderingContext2D>canvas?.getContext("2d");
  });

  it(`should draw a cells on canvas`, () => {
    const cells = [
      [0, 0, 0, 0, 0],
      [0, 1, 0, 0, 0],
      [0, 0, 0, 0, 0],
      [0, 0, 0, 0, 0],
      [0, 0, 0, 0, 0],
    ];
    drawGameOnCanvas("canvas", cells);
    const events = ctx.__getEvents();
    expect(events).toMatchSnapshot();
  });

  it(`should draw one white cell on canvas`, () => {
    const cells = [
      [1, 1, 1, 1, 1],
      [1, 0, 1, 1, 1],
      [1, 1, 1, 1, 1],
      [1, 1, 1, 1, 1],
      [1, 1, 1, 1, 1],
    ];
    drawGameOnCanvas("canvas", cells);
    const events = ctx.__getEvents();
    expect(events).toMatchSnapshot();
  });
});
