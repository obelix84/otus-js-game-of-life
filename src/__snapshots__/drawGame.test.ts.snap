// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Write cells on canvas should draw a cells on canvas 1`] = `
[
  {
    "props": {
      "height": 500,
      "width": 500,
      "x": 0,
      "y": 0,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  {
    "props": {},
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "beginPath",
  },
  {
    "props": {
      "value": "#808080",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillStyle",
  },
  {
    "props": {
      "height": 99,
      "width": 99,
      "x": 1,
      "y": 1,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  {
    "props": {
      "height": 99,
      "width": 99,
      "x": 101,
      "y": 1,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  {
    "props": {
      "height": 99,
      "width": 99,
      "x": 201,
      "y": 1,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  {
    "props": {
      "height": 99,
      "width": 99,
      "x": 301,
      "y": 1,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  {
    "props": {
      "height": 99,
      "width": 99,
      "x": 401,
      "y": 1,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  {
    "props": {
      "height": 99,
      "width": 99,
      "x": 1,
      "y": 101,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  {
    "props": {
      "fillRule": "nonzero",
      "path": [
        {
          "props": {
            "height": 99,
            "width": 99,
            "x": 101,
            "y": 101,
          },
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "rect",
        },
      ],
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fill",
  },
  {
    "props": {
      "height": 99,
      "width": 99,
      "x": 201,
      "y": 101,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  {
    "props": {
      "height": 99,
      "width": 99,
      "x": 301,
      "y": 101,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  {
    "props": {
      "height": 99,
      "width": 99,
      "x": 401,
      "y": 101,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  {
    "props": {
      "height": 99,
      "width": 99,
      "x": 1,
      "y": 201,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  {
    "props": {
      "height": 99,
      "width": 99,
      "x": 101,
      "y": 201,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  {
    "props": {
      "height": 99,
      "width": 99,
      "x": 201,
      "y": 201,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  {
    "props": {
      "height": 99,
      "width": 99,
      "x": 301,
      "y": 201,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  {
    "props": {
      "height": 99,
      "width": 99,
      "x": 401,
      "y": 201,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  {
    "props": {
      "height": 99,
      "width": 99,
      "x": 1,
      "y": 301,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  {
    "props": {
      "height": 99,
      "width": 99,
      "x": 101,
      "y": 301,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  {
    "props": {
      "height": 99,
      "width": 99,
      "x": 201,
      "y": 301,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  {
    "props": {
      "height": 99,
      "width": 99,
      "x": 301,
      "y": 301,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  {
    "props": {
      "height": 99,
      "width": 99,
      "x": 401,
      "y": 301,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  {
    "props": {
      "height": 99,
      "width": 99,
      "x": 1,
      "y": 401,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  {
    "props": {
      "height": 99,
      "width": 99,
      "x": 101,
      "y": 401,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  {
    "props": {
      "height": 99,
      "width": 99,
      "x": 201,
      "y": 401,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  {
    "props": {
      "height": 99,
      "width": 99,
      "x": 301,
      "y": 401,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  {
    "props": {
      "height": 99,
      "width": 99,
      "x": 401,
      "y": 401,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  {
    "props": {
      "value": "#000000",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillStyle",
  },
  {
    "props": {
      "height": 500,
      "width": 1,
      "x": 0,
      "y": 0,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillRect",
  },
  {
    "props": {
      "height": 500,
      "width": 1,
      "x": 100,
      "y": 0,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillRect",
  },
  {
    "props": {
      "height": 500,
      "width": 1,
      "x": 200,
      "y": 0,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillRect",
  },
  {
    "props": {
      "height": 500,
      "width": 1,
      "x": 300,
      "y": 0,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillRect",
  },
  {
    "props": {
      "height": 500,
      "width": 1,
      "x": 400,
      "y": 0,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillRect",
  },
  {
    "props": {
      "height": 1,
      "width": 500,
      "x": 0,
      "y": 0,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillRect",
  },
  {
    "props": {
      "height": 1,
      "width": 500,
      "x": 0,
      "y": 100,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillRect",
  },
  {
    "props": {
      "height": 1,
      "width": 500,
      "x": 0,
      "y": 200,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillRect",
  },
  {
    "props": {
      "height": 1,
      "width": 500,
      "x": 0,
      "y": 300,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillRect",
  },
  {
    "props": {
      "height": 1,
      "width": 500,
      "x": 0,
      "y": 400,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillRect",
  },
  {
    "props": {
      "height": 500,
      "width": 1,
      "x": 499,
      "y": 0,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillRect",
  },
  {
    "props": {
      "height": 1,
      "width": 500,
      "x": 0,
      "y": 499,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillRect",
  },
  {
    "props": {},
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "closePath",
  },
]
`;

exports[`Write cells on canvas should draw one white cell on canvas 1`] = `
[
  {
    "props": {
      "height": 500,
      "width": 500,
      "x": 0,
      "y": 0,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  {
    "props": {},
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "beginPath",
  },
  {
    "props": {
      "value": "#808080",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillStyle",
  },
  {
    "props": {
      "height": 99,
      "width": 99,
      "x": 1,
      "y": 1,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  {
    "props": {
      "height": 99,
      "width": 99,
      "x": 101,
      "y": 1,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  {
    "props": {
      "height": 99,
      "width": 99,
      "x": 201,
      "y": 1,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  {
    "props": {
      "height": 99,
      "width": 99,
      "x": 301,
      "y": 1,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  {
    "props": {
      "height": 99,
      "width": 99,
      "x": 401,
      "y": 1,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  {
    "props": {
      "height": 99,
      "width": 99,
      "x": 1,
      "y": 101,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  {
    "props": {
      "fillRule": "nonzero",
      "path": [
        {
          "props": {
            "height": 99,
            "width": 99,
            "x": 101,
            "y": 101,
          },
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "rect",
        },
      ],
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fill",
  },
  {
    "props": {
      "height": 99,
      "width": 99,
      "x": 201,
      "y": 101,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  {
    "props": {
      "height": 99,
      "width": 99,
      "x": 301,
      "y": 101,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  {
    "props": {
      "height": 99,
      "width": 99,
      "x": 401,
      "y": 101,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  {
    "props": {
      "height": 99,
      "width": 99,
      "x": 1,
      "y": 201,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  {
    "props": {
      "height": 99,
      "width": 99,
      "x": 101,
      "y": 201,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  {
    "props": {
      "height": 99,
      "width": 99,
      "x": 201,
      "y": 201,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  {
    "props": {
      "height": 99,
      "width": 99,
      "x": 301,
      "y": 201,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  {
    "props": {
      "height": 99,
      "width": 99,
      "x": 401,
      "y": 201,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  {
    "props": {
      "height": 99,
      "width": 99,
      "x": 1,
      "y": 301,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  {
    "props": {
      "height": 99,
      "width": 99,
      "x": 101,
      "y": 301,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  {
    "props": {
      "height": 99,
      "width": 99,
      "x": 201,
      "y": 301,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  {
    "props": {
      "height": 99,
      "width": 99,
      "x": 301,
      "y": 301,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  {
    "props": {
      "height": 99,
      "width": 99,
      "x": 401,
      "y": 301,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  {
    "props": {
      "height": 99,
      "width": 99,
      "x": 1,
      "y": 401,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  {
    "props": {
      "height": 99,
      "width": 99,
      "x": 101,
      "y": 401,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  {
    "props": {
      "height": 99,
      "width": 99,
      "x": 201,
      "y": 401,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  {
    "props": {
      "height": 99,
      "width": 99,
      "x": 301,
      "y": 401,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  {
    "props": {
      "height": 99,
      "width": 99,
      "x": 401,
      "y": 401,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  {
    "props": {
      "value": "#000000",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillStyle",
  },
  {
    "props": {
      "height": 500,
      "width": 1,
      "x": 0,
      "y": 0,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillRect",
  },
  {
    "props": {
      "height": 500,
      "width": 1,
      "x": 100,
      "y": 0,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillRect",
  },
  {
    "props": {
      "height": 500,
      "width": 1,
      "x": 200,
      "y": 0,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillRect",
  },
  {
    "props": {
      "height": 500,
      "width": 1,
      "x": 300,
      "y": 0,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillRect",
  },
  {
    "props": {
      "height": 500,
      "width": 1,
      "x": 400,
      "y": 0,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillRect",
  },
  {
    "props": {
      "height": 1,
      "width": 500,
      "x": 0,
      "y": 0,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillRect",
  },
  {
    "props": {
      "height": 1,
      "width": 500,
      "x": 0,
      "y": 100,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillRect",
  },
  {
    "props": {
      "height": 1,
      "width": 500,
      "x": 0,
      "y": 200,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillRect",
  },
  {
    "props": {
      "height": 1,
      "width": 500,
      "x": 0,
      "y": 300,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillRect",
  },
  {
    "props": {
      "height": 1,
      "width": 500,
      "x": 0,
      "y": 400,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillRect",
  },
  {
    "props": {
      "height": 500,
      "width": 1,
      "x": 499,
      "y": 0,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillRect",
  },
  {
    "props": {
      "height": 1,
      "width": 500,
      "x": 0,
      "y": 499,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillRect",
  },
  {
    "props": {},
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "closePath",
  },
  {
    "props": {
      "height": 500,
      "width": 500,
      "x": 0,
      "y": 0,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  {
    "props": {},
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "beginPath",
  },
  {
    "props": {
      "value": "#808080",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillStyle",
  },
  {
    "props": {
      "fillRule": "nonzero",
      "path": [
        {
          "props": {
            "height": 99,
            "width": 99,
            "x": 1,
            "y": 1,
          },
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "rect",
        },
      ],
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fill",
  },
  {
    "props": {
      "fillRule": "nonzero",
      "path": [
        {
          "props": {
            "height": 99,
            "width": 99,
            "x": 101,
            "y": 1,
          },
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "rect",
        },
      ],
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fill",
  },
  {
    "props": {
      "fillRule": "nonzero",
      "path": [
        {
          "props": {
            "height": 99,
            "width": 99,
            "x": 201,
            "y": 1,
          },
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "rect",
        },
      ],
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fill",
  },
  {
    "props": {
      "fillRule": "nonzero",
      "path": [
        {
          "props": {
            "height": 99,
            "width": 99,
            "x": 301,
            "y": 1,
          },
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "rect",
        },
      ],
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fill",
  },
  {
    "props": {
      "fillRule": "nonzero",
      "path": [
        {
          "props": {
            "height": 99,
            "width": 99,
            "x": 401,
            "y": 1,
          },
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "rect",
        },
      ],
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fill",
  },
  {
    "props": {
      "fillRule": "nonzero",
      "path": [
        {
          "props": {
            "height": 99,
            "width": 99,
            "x": 1,
            "y": 101,
          },
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "rect",
        },
      ],
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fill",
  },
  {
    "props": {
      "height": 99,
      "width": 99,
      "x": 101,
      "y": 101,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  {
    "props": {
      "fillRule": "nonzero",
      "path": [
        {
          "props": {
            "height": 99,
            "width": 99,
            "x": 201,
            "y": 101,
          },
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "rect",
        },
      ],
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fill",
  },
  {
    "props": {
      "fillRule": "nonzero",
      "path": [
        {
          "props": {
            "height": 99,
            "width": 99,
            "x": 301,
            "y": 101,
          },
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "rect",
        },
      ],
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fill",
  },
  {
    "props": {
      "fillRule": "nonzero",
      "path": [
        {
          "props": {
            "height": 99,
            "width": 99,
            "x": 401,
            "y": 101,
          },
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "rect",
        },
      ],
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fill",
  },
  {
    "props": {
      "fillRule": "nonzero",
      "path": [
        {
          "props": {
            "height": 99,
            "width": 99,
            "x": 1,
            "y": 201,
          },
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "rect",
        },
      ],
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fill",
  },
  {
    "props": {
      "fillRule": "nonzero",
      "path": [
        {
          "props": {
            "height": 99,
            "width": 99,
            "x": 101,
            "y": 201,
          },
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "rect",
        },
      ],
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fill",
  },
  {
    "props": {
      "fillRule": "nonzero",
      "path": [
        {
          "props": {
            "height": 99,
            "width": 99,
            "x": 201,
            "y": 201,
          },
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "rect",
        },
      ],
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fill",
  },
  {
    "props": {
      "fillRule": "nonzero",
      "path": [
        {
          "props": {
            "height": 99,
            "width": 99,
            "x": 301,
            "y": 201,
          },
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "rect",
        },
      ],
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fill",
  },
  {
    "props": {
      "fillRule": "nonzero",
      "path": [
        {
          "props": {
            "height": 99,
            "width": 99,
            "x": 401,
            "y": 201,
          },
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "rect",
        },
      ],
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fill",
  },
  {
    "props": {
      "fillRule": "nonzero",
      "path": [
        {
          "props": {
            "height": 99,
            "width": 99,
            "x": 1,
            "y": 301,
          },
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "rect",
        },
      ],
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fill",
  },
  {
    "props": {
      "fillRule": "nonzero",
      "path": [
        {
          "props": {
            "height": 99,
            "width": 99,
            "x": 101,
            "y": 301,
          },
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "rect",
        },
      ],
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fill",
  },
  {
    "props": {
      "fillRule": "nonzero",
      "path": [
        {
          "props": {
            "height": 99,
            "width": 99,
            "x": 201,
            "y": 301,
          },
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "rect",
        },
      ],
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fill",
  },
  {
    "props": {
      "fillRule": "nonzero",
      "path": [
        {
          "props": {
            "height": 99,
            "width": 99,
            "x": 301,
            "y": 301,
          },
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "rect",
        },
      ],
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fill",
  },
  {
    "props": {
      "fillRule": "nonzero",
      "path": [
        {
          "props": {
            "height": 99,
            "width": 99,
            "x": 401,
            "y": 301,
          },
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "rect",
        },
      ],
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fill",
  },
  {
    "props": {
      "fillRule": "nonzero",
      "path": [
        {
          "props": {
            "height": 99,
            "width": 99,
            "x": 1,
            "y": 401,
          },
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "rect",
        },
      ],
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fill",
  },
  {
    "props": {
      "fillRule": "nonzero",
      "path": [
        {
          "props": {
            "height": 99,
            "width": 99,
            "x": 101,
            "y": 401,
          },
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "rect",
        },
      ],
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fill",
  },
  {
    "props": {
      "fillRule": "nonzero",
      "path": [
        {
          "props": {
            "height": 99,
            "width": 99,
            "x": 201,
            "y": 401,
          },
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "rect",
        },
      ],
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fill",
  },
  {
    "props": {
      "fillRule": "nonzero",
      "path": [
        {
          "props": {
            "height": 99,
            "width": 99,
            "x": 301,
            "y": 401,
          },
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "rect",
        },
      ],
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fill",
  },
  {
    "props": {
      "fillRule": "nonzero",
      "path": [
        {
          "props": {
            "height": 99,
            "width": 99,
            "x": 401,
            "y": 401,
          },
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "rect",
        },
      ],
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fill",
  },
  {
    "props": {
      "value": "#000000",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillStyle",
  },
  {
    "props": {
      "height": 500,
      "width": 1,
      "x": 0,
      "y": 0,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillRect",
  },
  {
    "props": {
      "height": 500,
      "width": 1,
      "x": 100,
      "y": 0,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillRect",
  },
  {
    "props": {
      "height": 500,
      "width": 1,
      "x": 200,
      "y": 0,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillRect",
  },
  {
    "props": {
      "height": 500,
      "width": 1,
      "x": 300,
      "y": 0,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillRect",
  },
  {
    "props": {
      "height": 500,
      "width": 1,
      "x": 400,
      "y": 0,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillRect",
  },
  {
    "props": {
      "height": 1,
      "width": 500,
      "x": 0,
      "y": 0,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillRect",
  },
  {
    "props": {
      "height": 1,
      "width": 500,
      "x": 0,
      "y": 100,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillRect",
  },
  {
    "props": {
      "height": 1,
      "width": 500,
      "x": 0,
      "y": 200,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillRect",
  },
  {
    "props": {
      "height": 1,
      "width": 500,
      "x": 0,
      "y": 300,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillRect",
  },
  {
    "props": {
      "height": 1,
      "width": 500,
      "x": 0,
      "y": 400,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillRect",
  },
  {
    "props": {
      "height": 500,
      "width": 1,
      "x": 499,
      "y": 0,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillRect",
  },
  {
    "props": {
      "height": 1,
      "width": 500,
      "x": 0,
      "y": 499,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillRect",
  },
  {
    "props": {},
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "closePath",
  },
]
`;
